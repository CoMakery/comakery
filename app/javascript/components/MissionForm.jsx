import React from 'react'
import PropTypes from 'prop-types'
import {fetch as fetchPolyfill} from 'whatwg-fetch'
import InputFieldWhiteDark from './styleguide/InputFieldWhiteDark'
import InputFieldDescription from './styleguide/InputFieldDescription'
import InputFieldUploadFile from './styleguide/InputFieldUploadFile'
import Button from './styleguide/Button'
import ButtonBorder from './styleguide/ButtonBorder'
import Flash from './layouts/Flash'
import Layout from './layouts/Layout'

export default class MissionForm extends React.Component {
  constructor(props) {
    super(props)

    this.goBack = this.goBack.bind(this)
    this.errorAdd = this.errorAdd.bind(this)
    this.errorRemove = this.errorRemove.bind(this)
    this.disable = this.disable.bind(this)
    this.enable = this.enable.bind(this)
    this.handleChangeFormData = this.handleChangeFormData.bind(this)
    this.handleSaveMission = this.handleSaveMission.bind(this)
    this.verifyImageRes = this.verifyImageRes.bind(this)
    this.verifyLogoRes = this.verifyLogoRes.bind(this)

    this.imageInputRef = React.createRef()
    this.logoInputRef = React.createRef()

    this.state = {
      errors        : {}, // error hash for account form
      disabled      : {}, // disabled hash
      flashMessages : [],
      formAction    : this.props.formAction,
      formUrl       : this.props.formUrl,
      closeOnSuccess: false,
      name          : props.mission.name || '',
      subtitle      : props.mission.subtitle || '',
      description   : props.mission.description || '',
      logoPreview   : props.mission.logoUrl,
      imagePreview  : props.mission.imageUrl,
      logo          : null,
      image         : null
    }
    this.mounted = false
  }

  componentDidMount() {
    this.mounted = true
  }

  componentWillMount() {
    this.mounted = false
  }

  handleChangeFormData(e) {
    const target = e.target
    let value
    switch (target.type) {
      case 'checkbox':
        value = target.checked
        break
      case 'file':
        value = target.files[0]
        break
      default:
        value = target.value
    }
    const name = target.name
    this.setState({
      [name]: value,
    })

    if (!e.target.checkValidity()) {
      this.errorAdd(e.target.name, 'invalid value')
      return
    } else {
      this.errorRemove(e.target.name)
    }
  }

  handleSaveMission(e) {
    e.preventDefault()

    this.disable(['submit', 'submit_and_close'])

    if (!e.target.checkValidity()) {
      this.enable(['submit', 'submit_and_close'])
      return
    }

    let formData = new FormData()
    const {name, subtitle, description, logo, image} = this.state
    formData.append('mission[name]', name)
    formData.append('mission[subtitle]', subtitle)
    formData.append('mission[description]', description)
    formData.append('authenticity_token', this.props.csrfToken)
    if (logo) {
      formData.append('mission[logo]', logo)
    }
    if (image) {
      formData.append('mission[image]', image)
    }
    fetchPolyfill(this.state.formUrl, {
      credentials: 'same-origin',
      method     : this.state.formAction,
      body       : formData
    }).then(response => {
      if (response.status === 200) {
        if (this.state.closeOnSuccess) {
          window.location = this.props.urlOnSuccess
        } else {
          if (this.state.formAction === 'POST') {
            response.json().then(data => {
              this.setState(state => ({
                formAction   : 'PUT',
                formUrl      : `/missions/${data.id}`,
                flashMessages: state.flashMessages.concat([{'severity': 'notice', 'text': 'Mission Created'}])
              }))
              history.replaceState(
                {},
                document.title,
                `${window.location.origin}/missions/${data.id}/edit`
              )
            })
          } else {
            this.setState(state => ({
              flashMessages: state.flashMessages.concat([{'severity': 'notice', 'text': 'Mission Updated'}])
            }))
          }
          this.enable(['submit', 'submit_and_close'])
        }
      } else {
        response.json().then(data => {
          this.setState(state => ({
            errors       : data.errors,
            flashMessages: state.flashMessages.concat([{'severity': 'error', 'text': data.message}])
          }))
          this.enable(['submit', 'submit_and_close'])
        })
      }
    })
  }

  verifyImageRes(img) {
    if ((img.naturalWidth < 1200) || (img.naturalHeight < 800) || (img.naturalWidth / img.naturalHeight !== 1.5)) {
      this.imageInputRef.current.value = ''
      this.errorAdd('image', 'invalid resolution')
    } else {
      this.errorRemove('image')
    }
  }

  verifyLogoRes(img) {
    if ((img.naturalWidth < 800) || (img.naturalHeight < 800) || (img.naturalWidth / img.naturalHeight !== 1)) {
      this.logoInputRef.current.value = ''
      this.errorAdd('logo', 'invalid resolution')
    } else {
      this.errorRemove('logo')
    }
  }

  errorAdd(n, e) {
    this.setState({
      errors: Object.assign({}, this.state.errors, {[n]: e})
    })
  }

  errorRemove(n) {
    let e = this.state.errors
    delete e[n]
    this.setState({
      errors: e
    })
  }

  disable(a) {
    let d = this.state.disabled
    a.forEach(n => d = Object.assign({}, d, {[n]: true}))
    this.setState({
      disabled: d
    })
  }

  enable(a) {
    let d = this.state.disabled
    a.forEach(n => delete d[n])
    this.setState({
      disabled: d
    })
  }

  goBack() {
    window.location = '/missions'
  }

  render() {
    const {name, subtitle, description, logoPreview, imagePreview, errors} = this.state

    return <React.Fragment>
      <Layout
        className="mission-form"
        title={this.state.formAction === 'POST' ? 'Create a New Mission' : 'Edit Mission'}
        hasBackButton
        subfooter={
          <React.Fragment>
            <Button
              value={this.state.formAction === 'POST' ? 'create & close' : 'save & close'}
              type="submit"
              form="mission-form--form"
              className="button--right-space"
              disabled={this.state.disabled.submit_and_close}
              onClick={() => this.setState({closeOnSuccess: true})}
            />
            <ButtonBorder
              value={this.state.formAction === 'POST' ? 'create' : 'save'}
              type="submit"
              form="mission-form--form"
              className="button--right-space"
              disabled={this.state.disabled.submit}
              onClick={() => this.setState({closeOnSuccess: false})}
            />
            <ButtonBorder
              value="cancel"
              onClick={this.goBack}
            />
          </React.Fragment>
        }
      >

        <Flash messages={this.state.flashMessages} />

        <form id="mission-form--form" className="mission-form--form" onSubmit={this.handleSaveMission}>
          <InputFieldWhiteDark
            name="name" title="Name" symbolLimit={100} required value={name}
            placeholder="Unicoin"
            eventHandler={this.handleChangeFormData}
            errorText={errors.name} />

          <InputFieldWhiteDark
            name="subtitle" title="Subtitle" symbolLimit={140} required value={subtitle}
            placeholder="Unicoin is the future of tokenizing fantastical beast networks"
            eventHandler={this.handleChangeFormData}
            errorText={errors.subtitle} />

          <InputFieldDescription
            name="description" title="Description" symbolLimit={500} required value={description}
            placeholder="Here will be templated text but with lower opacity as lower opacity indicates that it is a placeholder. When user start to type within field, text should have 100% opacity."
            eventHandler={this.handleChangeFormData}
            errorText={errors.description}
          />

          <InputFieldUploadFile
            name="logo" title="Mission Logo" required
            imgPreviewDimensions="100x100"
            imgPreviewUrl={logoPreview}
            imgRequirements="Image should be at least 800px x 800px"
            imgVerifier={this.verifyLogoRes}
            imgInputRef={this.logoInputRef}
            eventHandler={this.handleChangeFormData}
            errorText={errors.logo} />
          <InputFieldUploadFile
            name="image" title="Mission Image" required
            imgPreviewDimensions="150x100"
            imgPreviewUrl={imagePreview}
            imgRequirements="Image should be at least 1200px x 800px"
            imgVerifier={this.verifyImageRes}
            imgInputRef={this.imageInputRef}
            eventHandler={this.handleChangeFormData}
            errorText={errors.image}
          />
        </form>
      </Layout>
    </React.Fragment>
  }
}

MissionForm.propTypes = {
  mission     : PropTypes.shape({}).isRequired,
  formUrl     : PropTypes.string.isRequired,
  formAction  : PropTypes.string.isRequired,
  urlOnSuccess: PropTypes.string.isRequired,
  csrfToken   : PropTypes.string.isRequired
}

MissionForm.defaultProps = {
  mission     : {},
  formUrl     : '/',
  formAction  : 'POST',
  urlOnSuccess: '/',
  csrfToken   : '00'
}
