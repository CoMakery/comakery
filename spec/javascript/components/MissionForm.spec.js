import React from 'react'
import { mount } from 'enzyme'
import MissionForm from 'components/MissionForm'

describe('MissionForm', () => {
  beforeEach(() => {
    fetch.resetMocks()
  })

  it('renders correctly without props', () => {
    const wrapper = mount(<MissionForm />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('Layout[title="Create a New Mission"]')).toBe(true)
    expect(wrapper.exists('.mission-form')).toBe(true)
    expect(wrapper.exists('ButtonBorder[value="cancel"]')).toBe(true)
    expect(wrapper.exists('ButtonBorder[value="create"]')).toBe(true)
    expect(wrapper.exists('Button[value="create & close"]')).toBe(true)
    expect(wrapper.exists('.mission-form--message')).toBe(false)
    expect(wrapper.exists('.mission-form--form')).toBe(true)
    expect(wrapper.exists('#mission-form--form')).toBe(true)

    expect(wrapper.exists(
      'InputFieldWhiteDark[title="Name"][symbolLimit=100][required][name="name"]'
    )).toBe(true)
    expect(wrapper.exists(
      'InputFieldWhiteDark[title="Subtitle"][required][name="subtitle"]'
    )).toBe(true)
    expect(wrapper.exists(
      'InputFieldDescription[title="Description"][required][name="description"]'
    )).toBe(true)
    expect(wrapper.exists(
      'InputFieldUploadFile[title="Mission Logo"][required][name="logo"]'
    )).toBe(true)
    expect(wrapper.exists(
      'InputFieldUploadFile[title="Mission Image"][required][name="image"]'
    )).toBe(true)
  })

  it('renders correctly with props', () => {
    const mission = {
      'id'         : 0,
      'name'       : 'Test Mission',
      'subtitle'   : 'test subtitle',
      'description': 'test description',
      'logoUrl'    : '/logo.png',
      'imageUrl'   : '/image.png'
    }
    const wrapper = mount(<MissionForm mission={mission} />)

    expect(wrapper.find(
      'InputFieldWhiteDark[title="Name"][symbolLimit=100][required][name="name"]'
    ).props().value).toBe('Test Mission')
    expect(wrapper.find(
      'InputFieldWhiteDark[title="Subtitle"][required][name="subtitle"]'
    ).props().value).toBe('test subtitle')
    expect(wrapper.find(
      'InputFieldDescription[title="Description"][required][name="description"]'
    ).props().value).toBe('test description')
    expect(wrapper.find(
      'InputFieldUploadFile[title="Mission Logo"][required][name="logo"]'
    ).props().imgPreviewUrl).toBe('/logo.png')
    expect(wrapper.find(
      'InputFieldUploadFile[title="Mission Image"][required][name="image"]'
    ).props().imgPreviewUrl).toBe('/image.png')
  })

  it('uses formUrl', () => {
    const wrapper = mount(<MissionForm formUrl="/test" />)

    expect(wrapper.state('formUrl')).toBe('/test')
  })

  it('uses formAction', () => {
    const wrapper = mount(<MissionForm formAction="PUT" />)

    expect(wrapper.state('formAction')).toBe('PUT')
  })

  it('uses urlOnSuccess', () => {
    const wrapper = mount(<MissionForm urlOnSuccess="/test" />)

    expect(wrapper.props().urlOnSuccess).toBe('/test')
  })

  it('displays messages', () => {
    const wrapper = mount(<MissionForm />)

    wrapper.setState({
      errorMessage: 'test error',
      infoMessage : 'test info'
    })

    wrapper.update()

    expect(wrapper.exists('Message[severity="error"][text="test error"]')).toBe(true)
    expect(wrapper.exists('Message[severity="warning"][text="test info"]')).toBe(true)
  })

  it('displays errors', () => {
    const wrapper = mount(<MissionForm />)

    wrapper.setState({
      errors: {
        'name'       : 'name error',
        'subtitle'   : 'subtitle error',
        'description': 'description error',
        'logo'       : 'logo error',
        'image'      : 'image error'
      }
    })

    wrapper.update()

    expect(wrapper.exists(
      'InputFieldWhiteDark[title="Name"][symbolLimit=100][required][name="name"][errorText="name error"]'
    )).toBe(true)
    expect(wrapper.exists(
      'InputFieldWhiteDark[title="Subtitle"][required][name="subtitle"][errorText="subtitle error"]'
    )).toBe(true)
    expect(wrapper.exists(
      'InputFieldDescription[title="Description"][required][name="description"][errorText="description error"]'
    )).toBe(true)
    expect(wrapper.exists(
      'InputFieldUploadFile[title="Mission Logo"][required][name="logo"][errorText="logo error"]'
    )).toBe(true)
    expect(wrapper.exists(
      'InputFieldUploadFile[title="Mission Image"][required][name="image"][errorText="image error"]'
    )).toBe(true)
  })
})
