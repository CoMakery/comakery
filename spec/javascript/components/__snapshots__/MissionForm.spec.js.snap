// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionForm renders correctly without props 1`] = `
<MissionForm
  csrfToken="00"
  formAction="POST"
  formUrl="/"
  mission={Object {}}
  urlOnSuccess="/"
>
  <Layout
    category=""
    className="mission-form"
    hasBackButton={true}
    navTitle={null}
    sidebar={null}
    subfooter={
      <React.Fragment>
        <Button
          className="button--right-space"
          disabled={false}
          form="mission-form--form"
          onClick={[Function]}
          type="submit"
          value="create & close"
        />
        <ButtonBorder
          className="button--right-space"
          form="mission-form--form"
          onClick={[Function]}
          type="submit"
          value="create"
        />
        <ButtonBorder
          className=""
          onClick={[Function]}
          value="cancel"
        />
      </React.Fragment>
    }
    title="Create a New Mission"
  >
    <div
      className="layout mission-form"
    >
      <div
        className="layout--back-button"
      >
        <Icon
          className=""
          name="iconCloseCopy.svg"
          onClick={[Function]}
        >
          <img
            className="icon icon__iconCloseCopy-svg"
            onClick={[Function]}
            src="test-file-stub"
          />
        </Icon>
      </div>
      <div
        className="layout--content"
      >
        <div
          className="layout--content--title"
        >
          Create a New Mission
        </div>
        <hr
          className="layout--content--hr"
        />
        <div
          className="layout--content--wrapper"
        >
          <div
            className="layout--content--content"
          >
            <Flash
              className=""
              messages={Array []}
            >
              <div
                className="flash-message-container"
              />
            </Flash>
            <form
              className="mission-form--form"
              id="mission-form--form"
              onSubmit={[Function]}
            >
              <InputFieldWhiteDark
                eventHandler={[Function]}
                name="name"
                placeholder="Unicoin"
                required={true}
                symbolLimit={100}
                title="Name"
                value=""
              >
                <InputField
                  checkboxText=""
                  checked={false}
                  className="input-field__white__dark"
                  copyOnClick={false}
                  disabled={false}
                  errorText=""
                  eventHandler={[Function]}
                  imgInputRef={null}
                  imgPreviewDimensions="40x40"
                  imgPreviewUrl=""
                  imgRequirements=""
                  imgVerifier={[Function]}
                  name="name"
                  pattern=".*"
                  placeholder="Unicoin"
                  readOnly={false}
                  recommended={false}
                  required={true}
                  selectEntries={Array []}
                  symbolLimit={100}
                  title="Name"
                  type="text"
                  value=""
                >
                  <div
                    className="input-field input-field__white__dark"
                  >
                    <div
                      className="input-field--title"
                    >
                      <span
                        className="input-field--title--title"
                      >
                        Name
                      </span>
                      <span
                        className="input-field--title--required"
                      >
                        required
                      </span>
                      <span
                        className="input-field--title--counter"
                      >
                        0
                        /
                        100
                      </span>
                    </div>
                    <div
                      className="input-field--content"
                    >
                      <input
                        className="input-field--content__text"
                        name="name"
                        onChange={[Function]}
                        onClick={null}
                        pattern=".*"
                        placeholder="Unicoin"
                        readOnly={false}
                        required={true}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </InputField>
              </InputFieldWhiteDark>
              <InputFieldWhiteDark
                eventHandler={[Function]}
                name="subtitle"
                placeholder="Unicoin is the future of tokenizing fantastical beast networks"
                required={true}
                symbolLimit={140}
                title="Subtitle"
                value=""
              >
                <InputField
                  checkboxText=""
                  checked={false}
                  className="input-field__white__dark"
                  copyOnClick={false}
                  disabled={false}
                  errorText=""
                  eventHandler={[Function]}
                  imgInputRef={null}
                  imgPreviewDimensions="40x40"
                  imgPreviewUrl=""
                  imgRequirements=""
                  imgVerifier={[Function]}
                  name="subtitle"
                  pattern=".*"
                  placeholder="Unicoin is the future of tokenizing fantastical beast networks"
                  readOnly={false}
                  recommended={false}
                  required={true}
                  selectEntries={Array []}
                  symbolLimit={140}
                  title="Subtitle"
                  type="text"
                  value=""
                >
                  <div
                    className="input-field input-field__white__dark"
                  >
                    <div
                      className="input-field--title"
                    >
                      <span
                        className="input-field--title--title"
                      >
                        Subtitle
                      </span>
                      <span
                        className="input-field--title--required"
                      >
                        required
                      </span>
                      <span
                        className="input-field--title--counter"
                      >
                        0
                        /
                        140
                      </span>
                    </div>
                    <div
                      className="input-field--content"
                    >
                      <input
                        className="input-field--content__text"
                        name="subtitle"
                        onChange={[Function]}
                        onClick={null}
                        pattern=".*"
                        placeholder="Unicoin is the future of tokenizing fantastical beast networks"
                        readOnly={false}
                        required={true}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </InputField>
              </InputFieldWhiteDark>
              <InputFieldDescription
                eventHandler={[Function]}
                name="description"
                placeholder="Here will be templated text but with lower opacity as lower opacity indicates that it is a placeholder. When user start to type within field, text should have 100% opacity."
                required={true}
                symbolLimit={375}
                title="Description"
                value=""
              >
                <InputField
                  checkboxText=""
                  checked={false}
                  className="input-field__description"
                  copyOnClick={false}
                  disabled={false}
                  errorText=""
                  eventHandler={[Function]}
                  imgInputRef={null}
                  imgPreviewDimensions="40x40"
                  imgPreviewUrl=""
                  imgRequirements=""
                  imgVerifier={[Function]}
                  name="description"
                  pattern=".*"
                  placeholder="Here will be templated text but with lower opacity as lower opacity indicates that it is a placeholder. When user start to type within field, text should have 100% opacity."
                  readOnly={false}
                  recommended={false}
                  required={true}
                  selectEntries={Array []}
                  symbolLimit={375}
                  title="Description"
                  type="textarea"
                  value=""
                >
                  <div
                    className="input-field input-field__description"
                  >
                    <div
                      className="input-field--title"
                    >
                      <span
                        className="input-field--title--title"
                      >
                        Description
                      </span>
                      <span
                        className="input-field--title--required"
                      >
                        required
                      </span>
                      <span
                        className="input-field--title--counter"
                      >
                        0
                        /
                        375
                      </span>
                    </div>
                    <div
                      className="input-field--content"
                    >
                      <textarea
                        className="input-field--content__text z"
                        name="description"
                        onChange={[Function]}
                        pattern=".*"
                        placeholder="Here will be templated text but with lower opacity as lower opacity indicates that it is a placeholder. When user start to type within field, text should have 100% opacity."
                        readOnly={false}
                        required={true}
                        value=""
                      />
                    </div>
                  </div>
                </InputField>
              </InputFieldDescription>
              <InputFieldUploadFile
                eventHandler={[Function]}
                imgInputRef={
                  Object {
                    "current": <input
                      accept="image/*"
                      class="input-field--content__file"
                      name="logo"
                      required=""
                      type="file"
                    />,
                  }
                }
                imgPreviewDimensions="100x100"
                imgRequirements="Image should be at least 800px x 800px"
                imgVerifier={[Function]}
                name="logo"
                required={true}
                title="Mission Logo"
              >
                <InputField
                  checkboxText=""
                  checked={false}
                  className="input-field__upload-file"
                  copyOnClick={false}
                  disabled={false}
                  errorText=""
                  eventHandler={[Function]}
                  imgInputRef={
                    Object {
                      "current": <input
                        accept="image/*"
                        class="input-field--content__file"
                        name="logo"
                        required=""
                        type="file"
                      />,
                    }
                  }
                  imgPreviewDimensions="100x100"
                  imgPreviewUrl=""
                  imgRequirements="Image should be at least 800px x 800px"
                  imgVerifier={[Function]}
                  name="logo"
                  pattern=".*"
                  placeholder="Please enter value"
                  readOnly={false}
                  recommended={false}
                  required={true}
                  selectEntries={Array []}
                  symbolLimit={0}
                  title="Mission Logo"
                  type="file"
                  value=""
                >
                  <div
                    className="input-field input-field__upload-file"
                  >
                    <div
                      className="input-field--title"
                    >
                      <span
                        className="input-field--title--title"
                      >
                        Mission Logo
                      </span>
                      <span
                        className="input-field--title--required"
                      >
                        required
                      </span>
                    </div>
                    <div
                      className="input-field--content"
                    >
                      <input
                        accept="image/*"
                        className="input-field--content__file"
                        name="logo"
                        onChange={[Function]}
                        required={true}
                        type="file"
                      />
                      <span
                        className="input-field--content__file--requirements"
                      >
                        Image should be at least 800px x 800px
                      </span>
                    </div>
                  </div>
                </InputField>
              </InputFieldUploadFile>
              <InputFieldUploadFile
                eventHandler={[Function]}
                imgInputRef={
                  Object {
                    "current": <input
                      accept="image/*"
                      class="input-field--content__file"
                      name="image"
                      required=""
                      type="file"
                    />,
                  }
                }
                imgPreviewDimensions="150x100"
                imgRequirements="Image should be at least 1200px x 800px"
                imgVerifier={[Function]}
                name="image"
                required={true}
                title="Mission Image"
              >
                <InputField
                  checkboxText=""
                  checked={false}
                  className="input-field__upload-file"
                  copyOnClick={false}
                  disabled={false}
                  errorText=""
                  eventHandler={[Function]}
                  imgInputRef={
                    Object {
                      "current": <input
                        accept="image/*"
                        class="input-field--content__file"
                        name="image"
                        required=""
                        type="file"
                      />,
                    }
                  }
                  imgPreviewDimensions="150x100"
                  imgPreviewUrl=""
                  imgRequirements="Image should be at least 1200px x 800px"
                  imgVerifier={[Function]}
                  name="image"
                  pattern=".*"
                  placeholder="Please enter value"
                  readOnly={false}
                  recommended={false}
                  required={true}
                  selectEntries={Array []}
                  symbolLimit={0}
                  title="Mission Image"
                  type="file"
                  value=""
                >
                  <div
                    className="input-field input-field__upload-file"
                  >
                    <div
                      className="input-field--title"
                    >
                      <span
                        className="input-field--title--title"
                      >
                        Mission Image
                      </span>
                      <span
                        className="input-field--title--required"
                      >
                        required
                      </span>
                    </div>
                    <div
                      className="input-field--content"
                    >
                      <input
                        accept="image/*"
                        className="input-field--content__file"
                        name="image"
                        onChange={[Function]}
                        required={true}
                        type="file"
                      />
                      <span
                        className="input-field--content__file--requirements"
                      >
                        Image should be at least 1200px x 800px
                      </span>
                    </div>
                  </div>
                </InputField>
              </InputFieldUploadFile>
            </form>
          </div>
        </div>
      </div>
      <div
        className="layout--subfooter"
      >
        <div
          className="layout--subfooter--buttons"
        >
          <Button
            className="button--right-space"
            disabled={false}
            form="mission-form--form"
            onClick={[Function]}
            type="submit"
            value="create & close"
          >
            <input
              className="button button--right-space"
              disabled={false}
              form="mission-form--form"
              onClick={[Function]}
              type="submit"
              value="create & close"
            />
          </Button>
          <ButtonBorder
            className="button--right-space"
            form="mission-form--form"
            onClick={[Function]}
            type="submit"
            value="create"
          >
            <Button
              className="button__border button--right-space"
              disabled={false}
              form="mission-form--form"
              onClick={[Function]}
              type="submit"
              value="create"
            >
              <input
                className="button button__border button--right-space"
                disabled={false}
                form="mission-form--form"
                onClick={[Function]}
                type="submit"
                value="create"
              />
            </Button>
          </ButtonBorder>
          <ButtonBorder
            className=""
            onClick={[Function]}
            value="cancel"
          >
            <Button
              className="button__border"
              disabled={false}
              onClick={[Function]}
              type="button"
              value="cancel"
            >
              <input
                className="button button__border"
                disabled={false}
                onClick={[Function]}
                type="button"
                value="cancel"
              />
            </Button>
          </ButtonBorder>
        </div>
      </div>
    </div>
  </Layout>
</MissionForm>
`;
